### PR TITLE
fix: Separate toasts storage to own MobX store

### LINKED ISSUE #2329: Separate toasts storage to own `MobX` store
It feels like 90% of the time the `UiStore` is imported into a component it is so that we can create a toast. Suggesting the following improvements for maintainability and performance:

- Add a hook for creating toasts (see API suggestion below).
- Use hook instead of access ui store in functional components. Note: there are still 7 class components where toasts are accessed, we can avoid converting to functional as part of this.
- Move storage of toasts to their own store, separate from `ui` – this allows ui store to be for interface settings only. This should be easy once the hook has abstracted away `ui.showToast` access

```javascript

import useToasts from "hooks/useToasts";

// within component
const { showToast, hideToast } = useToasts();

// usage as today
showToast(
  t("Content goest here"),
  { type: "info" }
);

```

### PR DESCRIPTION
**Pull request by iamsaumya**
closes #2329

### CODE REVIEW

**Comment by iamsaumya** [comment]
This is already too much to review. I will convert class components in some other PR 😄

File: app/hooks/useToasts.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    },
+    [toasts]
+  );
+
+  const hideToast = useCallback(
+    (toastId: string) => {
+      toasts.removeToast(toastId);
+    },
+    [toasts]
+  );
```
Comment (tommoor) [inline app/hooks/useToasts.js:23]: It doesn't seem like these memo's are needed, can pass the store methods directly back?

File: app/stores/ToastsStore.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    return id;
+  };
+
+  @action
+  removeToast = (id: string) => {
+    this.toasts.delete(id);
+  };
+
+  @computed
+  get orderedToasts(): Toast[] {
```
Comment (tommoor) [inline app/stores/ToastsStore.js:49]:
```suggestion
  get orderedData(): Toast[] {
```

Lets rename to match other stores now that it's single-purpose

File: app/stores/ToastsStore.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      type: options.type,
+      timeout: options.timeout,
+      action: options.action,
+    });
+    this.lastToastId = id;
+    return id;
+  };
+
+  @action
+  removeToast = (id: string) => {
```
Comment (tommoor) [inline app/stores/ToastsStore.js:44]: I'd be open to renaming this to `hideToast`

File: app/scenes/Document/components/Document.js [hunk 4]
```diff
@@ -91,7 +92,7 @@ class DocumentScene extends React.Component<Props> {
     } else if (prevProps.document.revision !== this.lastRevision) {
       if (auth.user && document.updatedBy.id !== auth.user.id) {
         this.props.toasts.showToast(
-          `Document updated by ${document.updatedBy.name}`,
+          t(`Document updated by ${document.updatedBy.name}`),
```
Comment (tommoor) [inline app/scenes/Document/components/Document.js:95]:
```suggestion
          t(`Document updated by {{userName}}`, { userName: document.updatedBy.name }),
```

Good catch, but gotta do the interpolation right 😉

Comment (iamsaumya) [reply app/scenes/Document/components/Document.js:95]: Ohh, sorry about that!

Comment (tommoor) [reply app/scenes/Document/components/Document.js]: I wrote it in on GitHub so sorry about the linting

**Review by tommoor — APPROVED** [review]

**Pull request merged**